rfcs/0004-pablo-distribution.adoc

```diff
@@ -0,0 +1,443 @@
+
+=== PBLO Stakers
+
+. System MUST allow staking of PBLO.
+. The system MUST accumulate the rewards share for PBLO holders who stake PBLO token, out of the PBLO supply allocated for them.
+. The system MUST support accumulating the (stakers) reward part of the Pablo trading fees.
+
+=== PICA Stakers
+
+. System MUST allow staking of PICA.
+. The system MUST accumulate the rewards share for PICA holders who stake PICA token, out of the PICA supply allocated for them.
+. The system MUST support accumulating any token rewards other than PICA for PICA stakers.
+
+=== Pablo Governance
+. Governance MUST be able to set the PBLO token reward allocation.
+. Governance MUST be able to set the Pablo LP reward proportion for each Pablo LP token(i.e Pool) out of PBLO or other token reward allocation. This is to incentivize providing liquidity to required pools as decided by governance.
+. Governance MUST be able to adjust the PBLO reward rate(eg: daily) based on the incentivization strategy.
+. Pablo pool protocol fees(for rewarding protocol stakers) SHOULD be configurable as a percentage of the pool owner fee.
+
+=== PICA Governance
+. Governance MUST be able to set the PICA token reward allocation.
+. Governance MUST be able to adjust the PICA reward rate based on the incentivization strategy.
+
+=== Technical Requirements
+. The system MUST allow accumulation and mapping of rewards shares of multiple assets types(Eg: PBLO, KSM) to staked position(fNFT) type defined by another asset type(eg: PICA).
+. The system MUST support transfer of rewards using staking-rewards pallet to necessary fNFT types.
+. The system SHOULD support converting a reward accumulated in one asset type to another based on a preferred reward asset type configuration. Eg: Given a reward accumulated is in Acala it should be able to convert that to one of PBLO or PICA using the Pablo DEX pools.
+** This is to handle cases where a Pablo pool fees are in a different asset type than what is preferred.
+
+== Method
+
+=== System Overview
+
+[plantuml,images/pablo-distribution-verview,png]
+----
+skinparam component {
+  backgroundColor<<exists>> LightGreen
+}
+
+node "Pallet-Staking-Rewards\n(Pallet-Rewards?)" {
+    StakingReward - [Reward Storage/Logic]
+    [RewardsConfig]
+    [RewardsConfig] --> [Reward Storage/Logic]
+    Staking - [fNFT Storage/Logic]
+    Node "BatchProcess" {
+        [RewardDisbursementHook]
+        [RewardsConfig] --> [RewardAccumulationHook]
+    }
+}
+
+[Reward Storage/Logic] <<exists>>
+[fNFT Storage/Logic] <<exists>>
+[RewardDisbursementHook] <<exists>>
+
+node "Pallet-Pablo" {
+    [Pool] --> StakingReward : transfer\ntrading fees\n(for LPs and PBLO stakers)
+    [Pool] --> Staking : stake LP tokes
+    [FeeConfig] --> [Pool]
+}
+
+[Pool] <<exists>>
+
+node "Governance" {
+    [Configuration] --> [RewardsConfig]
+    [Configuration] --> [FeeConfig]
+}
+----
+
+TODO: Suggest rename of staking-reward pallet to pallet-earn or pallet-rewards.
```

Comment:
may be we can rename later when have more rewards mechanics, not now

Reply:
The reason to suggest this is that with this design we would already be using the staking-rewards pallet to calculate and disburse LP fees. So it makes sense to have a more generic name like `pallet-earn`. @hussein-aitlahcen any ideas?

Reply:
let look if we really has 2 composable pallets, actually whole composable is `earn` runtime. so i would not be so broad.